### ISSUE TITLE
import_plugin hides exception source

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [pytest-dev/pytest#2491]
```
Traceback (most recent call last):
  File "/home/rpfannsc/Projects/pytest-dev/pytest-venv/bin/pytest", line 11, in <module>
    load_entry_point('pytest', 'console_scripts', 'pytest')()
  File "/home/rpfannsc/Projects/pytest-dev/pytest-venv/lib/python3.5/site-packages/pkg_resources/__init__.py", line 560, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/home/rpfannsc/Projects/pytest-dev/pytest-venv/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2648, in load_entry_point
    return ep.load()
  File "/home/rpfannsc/Projects/pytest-dev/pytest-venv/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2302, in load
    return self.resolve()
  File "/home/rpfannsc/Projects/pytest-dev/pytest-venv/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2308, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/rpfannsc/Projects/pytest-dev/pytest/pytest.py", line 76, in <module>
    _preloadplugins()  # to populate pytest.* namespace so help(pytest) works
  File "/home/rpfannsc/Projects/pytest-dev/pytest/_pytest/config.py", line 117, in _preloadplugins
    _preinit.append(get_config())
  File "/home/rpfannsc/Projects/pytest-dev/pytest/_pytest/config.py", line 126, in get_config
    pluginmanager.import_plugin(spec)
  File "/home/rpfannsc/Projects/pytest-dev/pytest/_pytest/config.py", line 444, in import_plugin
    raise new_exc
ImportError: Error importing plugin "unittest": cannot import name 'fail'

```
as an example

its not that helpful to get the traceback cut at a bad location

**Comment by srinivasreddy** [comment]
I am taking it. Or somebody working on this?

**Comment by nicoddemus** [comment]
Not that I know of, please go ahead! :+1:

**Comment by srinivasreddy** [comment]
@RonnyPfannschmidt Could you please let me know what kind of stack-trace you are expecting?

**Comment by RonnyPfannschmidt** [comment]
@srinivasreddy anything that includes the origin of the error, python3 does this nicely on its own when correclty chaining exceptions,
i wonder if a decraded mode for python3 is justifiable due to unreasonable hardship

**Comment by nicoddemus** [comment]
Now that we use `six`, does `six.reraise` provides better results I wonder?

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus it cant magically enhance python2 as far as i understood

**Comment by nicoddemus** [comment]
I was thinking of something like this:

```python
import six, sys


def load_myplugin():
    from unittest import fail

def load_all_plugins():
    try:
        load_myplugin()
    except Exception as e:
        if six.PY2:
            six.reraise(type(e), type(e)('While loading plugin "myplugin":' + '\n' + str(e)), sys.exc_info()[2])
        else:
            raise type(e)('While loading plugin myplugin')

load_all_plugins()
```

Python 2:

```
Traceback (most recent call last):
  File "foo.py", line 16, in <module>
    load_all_plugins()
  File "foo.py", line 12, in load_all_plugins
    six.reraise(type(e), type(e)('While loading plugin "myplugin":' + '\n' + str(e)), sys.exc_info()[2])
  File "foo.py", line 9, in load_all_plugins
    load_myplugin()
  File "foo.py", line 5, in load_myplugin
    from unittest import fail
ImportError: While loading plugin "myplugin":
cannot import name fail
```

Python 3:

```
Traceback (most recent call last):
  File "foo.py", line 9, in load_all_plugins
    load_myplugin()
  File "foo.py", line 5, in load_myplugin
    from unittest import fail
ImportError: cannot import name 'fail'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "foo.py", line 16, in <module>
    load_all_plugins()
  File "foo.py", line 14, in load_all_plugins
    raise type(e)('While loading plugin myplugin')
ImportError: While loading plugin myplugin
```

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus those combinations look good

**Comment by OfirOshir** [comment]
I created a pr to solve this issue can someone plz check it?
https://github.com/pytest-dev/pytest/pull/2762

**Comment by nicoddemus** [comment]
Fixed by #2791
